### DISCUSSION TITLE
Nondeterministic Monolith Setup

### DISCUSSION THREAD

**Discussion by tawaren** [JanssenProject/jans#9871] [Q&A] [answered]
I currently use the docker monolith for development and testing.
Multiple times now, I had the situation that after a fresh setup (`clean.sh` followed by `run.sh`), something changed even if I did not alter the used Janssen monolith image version. I assume this is because the install script downloads something that is not versioned. As the installer `jans-linux-setup/jans_setup/install.py` is shared with the Linux setup, it may behave similarly. 

I know that the monolith is development & test use only, and this may be the expected (or an accepted) behaviour; in that case, I can live with that. I just wanted to mention it in case this is not how it should work.

**Comment by ossdhaval** [comment]
@tawaren You are right, docker monolith is for development and hence it might be pulling images from nightly build. We will confirm on this. Thanks for bringing this up.

**Comment by moabu** [comment, marked as answer]
Hey @tawaren ,

Yes that is expected. So long you use the one from main that will be the case. If you go to the versioned it should pull the latest from that version. 

With that said the docker-monolith is just an encapsulation of the jans linux setup and meant purely for a quick testing and running tests off of.
